EpsilonDriverControls/Tests/tests/DriverControlsTest/DriverControlsTest.c

```diff
@@ -0,0 +1,926 @@
+                              "Expected driveCommandsInfo.prevResetStatus not 1");
+    TEST_ASSERT_EQUAL_MESSAGE(0, expectedDriveCommandsInfo.motorCurrentOut,
+                              "Expected driveCommandsInfo.motorCurrentOut not 0");
+
+    //memcpy copies 8 bytes of data into msg.data and msg.data stores 8 bits of data in each index.
+    uint8_t expectedMsgData[8] = {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0};
+    TEST_ASSERT_EQUAL_UINT8_ARRAY_MESSAGE(expectedMsgData, expectedMotorDriveMsg.Data, 8,
+                                          "Expected motor drive msg.Data context or size incorrect");
+}
+
+//Helper funciton to check the motor drive CAN msg->StdId and DLC.
+void checkDriveMsgStdidAndDlcForDriveCommands(CanMsg* expectedMotorDriveMsg)
+{
+    TEST_ASSERT_EQUAL_MESSAGE(MOTOR_DRIVE_STDID, expectedMotorDriveMsg->StdId,
+                              "Expected motor drive msg.StdId not MOTOR_DRIVE_STDID");
+    TEST_ASSERT_EQUAL_MESSAGE(MOTOR_DRIVE_DLC, expectedMotorDriveMsg->DLC,
+                              "Expected motor drive msg.DLC not MOTOR_DRIVE_DLC");
+}
+
+//Helper function to check motor power CAN message things as they do not change.
+void checkPowerMessageForDriveCommands(CanMsg* expectedMotorPowerMsg)
+{
+    uint8_t expectedMsgData[8] = {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x80, 0x3f};
+    TEST_ASSERT_EQUAL_MESSAGE(MOTOR_POWER_STDID, expectedMotorPowerMsg->StdId,
+                              "Expected motor power msg.StdId not MOTOR_POWER_STDID");
+    TEST_ASSERT_EQUAL_MESSAGE(MOTOR_POWER_DLC, expectedMotorPowerMsg->DLC,
+                              "Expected motor power msg.DLC not MOTOR_POWER_DLC");
+    TEST_ASSERT_EQUAL_UINT8_ARRAY_MESSAGE(expectedMsgData, expectedMotorPowerMsg->Data, 8,
+                                          "Expected motor power msg.Data context or size incorrect");
+}
+
+//helper function to set regen values queue
+void setRegenValuesQueue(uint8_t number)
+{
+    for (int i = 0; i < REGEN_QUEUE_SIZE; i++)
+    {
+        regenValuesQueue[i] = number;
+    }
+}
+
+//helper function to set accel values queue
+void setAccelValuesQueue(uint8_t number)
+{
+    for (int i = 0; i < ACCEL_QUEUE_SIZE; i++)
+    {
+        accelValuesQueue[i] = number;
+    }
+}
+
+//helper function for osPoolAlloc and osMessagePut expect and returns for motor drive message
+void checkExpectMotorDriveMessage(CanMsg* expectedMotorDriveMsg)
+{
+    osPoolAlloc_ExpectAndReturn(canPool, expectedMotorDriveMsg);
+    osMessagePut_ExpectAndReturn(canQueue, (uint32_t)expectedMotorDriveMsg, osWaitForever, osOK);
+}
+
+//helper function for osPoolAlloc and osMessagePut expect and returns for motor power message
+void checkExpectMotorPowerMessage(CanMsg* expectedMotorPowerMsg)
+{
+    osPoolAlloc_ExpectAndReturn(canPool, expectedMotorPowerMsg);
+    osMessagePut_ExpectAndReturn(canQueue, (uint32_t)expectedMotorPowerMsg, osWaitForever, osOK);
+}
```

Comment:
These two functions are pretty much the same thing, can one `expectCanMessage` function be used for both? then it can also be used wit the other `expected` CanMsgs throughout the file.